solr/solrj/src/java/org/apache/solr/client/solrj/io/eval/ListCacheEvaluator.java

```diff
@@ -41,35 +39,21 @@ public ListCacheEvaluator(StreamExpression expression, StreamFactory factory) th
   }
 
   @Override
-  @SuppressWarnings({"unchecked"})
   public Object doWork(Object... values) throws IOException {
-    @SuppressWarnings({"rawtypes"})
-    ConcurrentMap objectCache = this.streamContext.getObjectCache();
-    @SuppressWarnings({"rawtypes"})
-    List list = new ArrayList();
+    Map<String, Object> objectCache = this.streamContext.getObjectCache();
+    List<String> list = new ArrayList<>();
 
     if(values.length == 0) {
-      @SuppressWarnings({"rawtypes"})
-      ConcurrentHashMap m = (ConcurrentHashMap)objectCache;
-      @SuppressWarnings({"rawtypes"})
-      Enumeration en = m.keys();
-      while(en.hasMoreElements()) {
-        list.add(en.nextElement());
-      }
+      list.addAll(objectCache.keySet());
       return list;
     } else if(values.length == 1) {
       String space = (String)values[0];
       space = space.replace("\"", "");
-      @SuppressWarnings({"rawtypes"})
-      ConcurrentMap spaceCache = (ConcurrentMap)objectCache.get(space);
+      Map<?,?> spaceCache = (Map<?,?>)objectCache.get(space);
       if(spaceCache != null) {
-        @SuppressWarnings({"rawtypes"})
-        ConcurrentHashMap spaceMap = (ConcurrentHashMap)objectCache.get(space);
-        @SuppressWarnings({"rawtypes"})
-        Enumeration en = spaceMap.keys();
-        while(en.hasMoreElements()) {
-          list.add(en.nextElement());
-        }
+        @SuppressWarnings("unchecked")
+        Map<String,?> spaceMap = (Map<String,?>)objectCache.get(space);
```

Comment:
It's not what the existing code does but would

```suggestion
        Map<String,?> spaceMap = (Map<String,?>)spaceCache;
```

be equivalent (and clearer) here? And how do we know the map key is a string (though it seems likely given the `Map<Object, Object> Tuple.fields` type)?